content/backend/graphql-scala/1-getting-started.md

```diff
@@ -0,0 +1,214 @@
+
+```
+
+</Instruction>
+
+### HTTP Server
+
+Time to create HTTP Server
+
+<Instruction>
+
+Create `Server.scala` file which will be our HTTP server and entry point for the application.
+Use content as follows:
+
+```scala
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.server.Route
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.server.Directives._
+import spray.json._
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+
+import scala.concurrent.Await
+import scala.language.postfixOps
+
+//1
+object Server extends App {
+
+  //2
+  val PORT = 8080
+
+  implicit val actorSystem = ActorSystem("graphql-server")
+  implicit val materializer = ActorMaterializer()
+
+  import actorSystem.dispatcher
+  import scala.concurrent.duration._
+
+  scala.sys.addShutdownHook(() -> shutdown())
+
+  //3
+  val route: Route = {
+    complete("Hello GrahpQL Scala!!!")
+  }
+
+  Http().bindAndHandle(route, "0.0.0.0", PORT)
+  println(s"open a browser with URL: http://localhost:$PORT")
+
+
+  def shutdown(): Unit = {
+    actorSystem.terminate()
+    Await.result(actorSystem.whenTerminated, 30 seconds)
+  }
+}
+```
+
+</Instruction>
+
+Our server extends an `App` trait so SBT can find it and run when you'll use `sbt run` command. When there are more such files in your project, SBT will ask you which one you want to run.
+
+At point 2, there is defined port number we want to use, choose what you want if proposed `8080` doesn't work for you. The main point of the file is `val route` definitions. For now, in all cases, server responds with simple text to prove it's working, but no worries, you will change this value in the near future.
+
+What is worth to point out here: In our example I use [Spray JSON](https://github.com/spray/spray-json) library for marshalling and unmarshalling JSON objects, but it isn't obligatory for you. You can use whatever JSON library you want, but in such case you have to change dependencies. [On this page](http://sangria-graphql.org/download/) you can find what JSON libraries Sagria can play with.
+
+### Database configuration
+
+In our project I chose to use H2 database. It's easy to configure and is able to run in memory - you no need to install any additional pakcages in your OS. In such cases like this tutorial H2 works perfectly, but if want to use another DB, it's up to you, Slick supports many of them.
+
+<Instruction>
+
+Inside `src/main/resources` directory, create an `application.conf` file with the following content:
+
+```
+h2mem = {
+  url = "jdbc:h2:mem:howtographqldb"
+  driver = org.h2.Driver
+  connectionPool = disabled
+  keepAliveConnection = true
+}
+```
+
+</Instruction>
+
+It's all we need to configure a Database, now we're ready to use it. For the future purposes we will create two additional files.
+The first `DAO` object will be responsible for accessing database. We will put there all the functions responsible for managing database data.
```

Comment:
`database data` sounds like masło maślane, maybe `for managing our data in the DB`?

Reply:
`The first one will be a \`DAO\` object, which will be responsible (...)` - we only mentioned two files and you now say it's `the first dao` - which implies there are more (and you didn't mention the files will include a dao) - but I might be over sensitive here